ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/sflow.schema.yml

```
@@ -60,9 +65,24 @@ keys:
               display_name: Port Number
               convert_types:
               - str
+      source:
+        type: str
+        display_name: Source IP
       source_interface:
         type: str
         display_name: Source Interface
+      extensions:
+        type: list
+        primary_key: name
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              description: Extension Name
+            enabled:
+              type: bool
+              description: Enable or Disable Extension
```

Your logic in the eos template only works if this is set. So either set enabled as required _or_ set a `default: true` and adjust the logic.
```suggestion
              description: Enable or Disable Extension
              required: true
```
FYI the primary key is always `required: true` implicitly, so it is up to you if you also want to add it there.

---

I'm setting "enabled" as required. I think the user should explicitly enable or disable these extensions